lightning/src/ln/peer_handler.rs

```diff
@@ -3191,4 +3213,100 @@ mod tests {
 		thread_c.join().unwrap();
 		assert!(cfg[0].chan_handler.message_fetch_counter.load(Ordering::Acquire) >= 1);
 	}
+
+	#[test]
+	#[cfg(feature = "std")]
+	fn test_rapid_connect_events_order_multithreaded() {
```

Comment:
I ran this about 100x with your change reverted and wasn't able to get it to fail :/